numba/core/config.py

```diff
@@ -80,6 +80,42 @@ def _validate_captured_errors_style(style_str):
         return rendered_style
 
 
+class _OptLevel(int):
+    """This class holds the "optimisation level" set in `NUMBA_OPT`. As this env
+    var can be an int or a string, but is almost always interpreted as an int.
+    This class subclasses int so as to get the common behaviour but stores the
+    actual value as a `_raw_value` member to make it available for cases where
+    accounting for the specific string supplied by the user is necessary."""
+
+    def __new__(cls, *args, **kwargs):
+        assert len(args) == 1
+        (value,) = args
+        _int_value = 3 if value == 'max' else int(value)
+        # the int ctor is always called with an appropriate integer value
+        new = super().__new__(cls, _int_value, **kwargs)
+        # raw value is max or int
+        new._raw_value = value if value == 'max' else _int_value
```

Comment:
This class can have a `is_max_opt` method/property so codegen doesn't need to access this attribute.

Reply:
Thanks, I'll make this change because it's preferable to not access the attribute directly, _but_, if in future for some reason we have to add numerous special cases, this may need revisiting as an approach.

Reply:
Implemented in c91b102.